solutions/standard/DA-types.md

```diff
@@ -0,0 +1,145 @@
+# IBM Cloud Elasticsearch Deployable Architecture Variables Documentation
+This document provides detailed information about the complex object types used in the IBM Cloud Elasticsearch DA (Deployable Architecture).
+
+
+## 1. Service Credential Names
+
+**Description:**
+This variable defines a map of service credential names and their corresponding roles for the Elasticsearch database. This allows for the creation of service credentials with specific roles, providing controlled access to the database.
+Reference provided for [Account Service Credentials](https://cloud.ibm.com/docs/account?topic=account-service_credentials&interface=ui).
+
+**Type:**
+The `service_credential_names` variable is a map where the key is the name of the service credential and the value is the role assigned to that credential.
+
+**Default Value:**
+The default value for the `service_credential_names` variable is an empty map (`{}`).
+
+- `Key (name, required):` The name of the service credential.
+
+- `Value (role, required):` The role assigned to the service credential.
+
+**Sample Examples:**
+
+```json
+  {
+    "elastic_user_1" = "Viewer"
+  }
+```
+```json
+  {
+    "elastic_user_2" = "Editor"
+  }
+```
+
+## 2. Users
+
+**Description:**
+
+This variable defines a list of users who have access to the Elasticsearch database. Each user in the list includes a name, password, type, and an optional role. This configuration creates native Elasticsearch database users.
+
+**Type:**
+The `users` variable is a list of objects, where each object represents a user with the following properties:
+ - `name (required):` The username for the user account.
+
+ - `password (required):` The password for the user account. This password must be in the range of 10-32 characters. This field is sensitive and should be handled securely.
+
+ - `type (required)`: This is to specify the type of user. The "type" field is required to generate the connection string for the outputs.
+
+ - `role (optional):` Defines the role assigned to the user, determining their access level and permissions.
+
+
+**Default Value:**
+The default value for the `users` variable is an empty list (`[]`).
+
+**Sensitive:**
+The `users` variable is marked as sensitive due to the inclusion of passwords.
+
+**Sample Example:** The below example shows the list of two users where one of the users is not provided the optional `role` value.
+```json
+[
+    {
+      name     = "es_admin"
+      password = "securepassword123" # pragma: allowlist secret
+      type     = "admin"
+      role     = "admin"
+    },
+    {
+      name     = "es_reader"
+      password = "readpassword123" # pragma: allowlist secret
```

Comment:
remove `# pragma: allowlist secret` and instead update the secrets baseline:
```
detect-secrets scan --update .secrets.baseline
detect-secrets audit .secrets.baseline
```

Reply:
done